doc: west: Document build system integration (#12938)
zephyrproject-rtos/zephyr: doc/tools/west/repo-tool.rst

  that fetches upstream data is run (this behavior can be
  suppressed for the duration of a single command by passing ``--no-update``).

.. _west-mr-build:

Build system integration
************************

> Some of the projects managed by west require to be integrated with the Zephyr

Reviewer: ```suggestion
Some of the projects managed by west must be integrated with the Zephyr
```

build system, which is based around Kconfig and CMake (see

> :ref:`application` for more information about Zephyr's build system).

Reviewer 2: Suggest adding:

Examples include vendor HALs and other firmware support libraries.

The way that the build system discovers and includes west-managed projects when
building an application is based around the ``west list`` command. The main

> CMake script in zephyr (the :file:`CMakeLists.txt` located at the zephyr

Reviewer: Zephyr should be capitalized when it's a reference to the project or OS (and not a directory name).

Reviewer 2: In this case it is referring to the "zephyr repository root folder", so it is a directory name.

> repository root folder) invokes ``west list`` in order to obtain a list of

Reviewer: ```suggestion
repository root folder) invokes ``west list`` to obtain a list of
```

projects managed by west.

> The CMake code in :file:`CMakeLists.txt` retrieves the following

Reviewer 3: Just a small question, would you also had written `The C code in file.c` or `The Python code in file.py` ?
If not, I would suggest just: ```The code in :file: `CMakeLists.txt` ```

information for each project using ``west list``:

> - name: The name of the project as specified in the manifest file

Reviewer: need a blank line before the first item of a bullet list

- path: The path of the project within the west installation

Once it has collected the list, the CMake script performs the following
operations on each project:

> - If a matching :file:`CMakeLists.txt` is located, it will process it directly

Reviewer 2: `zephyr/CMakeLists.txt` and `zephyr/Kconfig`, right? Or did you change this decision after I left the meeting?

Reviewer 2: Also, this seems a bit redundant with the below paragraph that says the same thing.

Author: I will rework this because it is clearly confusing

  by using the CMake ``add_subdirectory`` command
- If a matching :file:`Kconfig` is located, it will instruct Kconfig to source
  it (i.e. process it)

The way that the script determines if a matching :file:`CMakeLists.txt` and
:file:`Kconfig` are present is the following (in order of precedence):

> - If the project contains a file named :file:`<path>/zephyr/project.yml` then

Reviewer 3: as stated in https://github.com/zephyrproject-rtos/zephyr/pull/12938#issuecomment-459638161 , I prefer `zephyr/module.yml`

Author: I'll add a comment below to discuss the names

Reviewer 3: Great. I only placed this comment here, so that it would be easy to find the location of `zephyr/project.yml` later.

  its contents are parsed and the following fields are processed if present:

>   - ``cmake:`` points to a folder containing the :file:`CMakeLists.txt` that

Reviewer: Need a blank line before the first bullet list item (in this nested list)
